backend/internal/handlers/webhooks/webhooks.go

```diff
@@ -99,10 +90,26 @@ func (s *WebHookService) baseRepoInitialization(c *fiber.Ctx, pushEvent github.P
 		return errs.BadRequest(errors.New("invalid repository data"))
 	}
 
-	// Create necessary repo branches
+	// Retrieve assignment deadline from DB
+	deadline, err := s.store.GetAssignmentDueDateByRepoName(c.Context(), *pushEvent.Repo.Name)
+	if err == nil {
+		// There is a deadline
```

Comment:
This is a faulty way to look for a deadline. If another error happens, the code will mistakenly proceed.

IMO, you're better off defining a function `GetAssignmentByBaseRepoName` and handling the logic for whether there is/isn't a due date here.

Reply:
✅